packages/client/src/api/wargames_api/index.ts

```diff
@@ -682,36 +682,19 @@ export const postFeedback = (dbName: string, fromDetails: MessageDetailsFrom, tu
   return db.put(feedback).catch(rejectDefault)
 }
 
-const checkReference = (message: MessageCustom, db: ApiWargameDb, details: MessageDetails) => {
+const checkReference = (message: MessageCustom, db: ApiWargameDb, details: MessageDetails): Promise<MessageCustom> => {
   // eslint-disable-next-line no-async-promise-executor
   return new Promise(async (resolve): Promise<void> => {
     if (message.details.messageType !== 'Chat') {
       // default value for message counter
+      
```

Comment:
@lilitkarapetyan - I'm sure we only need to do this if `message.message.Reference` is present.